### ISSUE TITLE
garmin connect not syncing again...

### ISSUE THREAD

**Issue by wojtek-b** [cpfair/tapiriik#273]
Just to let you know...

**Comment by RobDickson** [comment]
No Garmin activities since 11/12/16 appear in my list of activities. I haven't done much exercise this week, but at least 6 activities should be there.

**Comment by RobDickson** [comment]
6 more Garmin activities, but still none shown in my Tapiriik list of activities since 11/12/16. Any idea when this will be fixed?

**Comment by meeximum** [comment]
Maybe same Problem as #261 
Solution would be to get data via download TCX api, but I didn't get it working :-/

**Comment by RobDickson** [comment]
I thought that Collin had tweeted to say that the bug mentioned in #261 had been fixed.

**Comment by cpfair** [comment]
As it turns out, there was an issue deploying the change that fixed
https://github.com/cpfair/tapiriik/issues/278, so the system wouldn't have
been picking up new activities in existing GC accounts. I think that's
fixed now.

**Comment by RobDickson** [comment]
Thanks. It worked for a couple of days, but seems to have stopped working again. I'll try a manual sync now.

**Comment by RobDickson** [comment]
Working again now. It seems my auto sync was stopped earlier than I expected. I've renewed it now.

Thanks again.

**Comment by RobDickson** [comment]
Two activities on 27/12/16 and two activities on 29/12/16 are shown in Garmin Connect, but none of these is shown on my activities in Tapiriik. Automatic sync is active and I've tried a manual sync.

I know it's Christmas, but do you think this will be fixed soon?

I'm sorry, I'm not a coder, so can't offer to help.

**Comment by RobDickson** [comment]
Suddenly it started working again yesterday morning. Thanks to whoever fixed it.

**Comment by RobDickson** [comment]
It's stopped working again now.

**Comment by RobDickson** [comment]
It started working again this morning. Thanks to whoever fixed it this time. Hopefully it'll be OK for a while now.

**Comment by mishamsk** [comment]
Stopped syncing today...Same problem?

**Comment by RobDickson** [comment]
It's the same for me. It synchronised OK yesterday, but not today.

**Comment by mishamsk** [comment]
I've been testing tapirik the day before yesterday and everything and it went through 174 activity without any problems twice, synchronized almost all of them. Yesterday there were some problems with the Garmin connect itself, activity list was not working and service status dashboard indicated some issues. However by the time of my comment everything was fine on Garmin side... Is there a way to see sync logs to investigate?

**Comment by RobDickson** [comment]
At the moment, it's working, just very slowly.

Sunday's swim took about 8 hours to appear in Strava.

Monday's ride took about 20 hours to appear in Strava.

When I first started using Tapiriik just over a year ago, it often took several hours for activities to appear in Strava, but it's been much better up until it broke last month.

**Comment by mishamsk** [comment]
@RobDickson so I've set up a "local" tapiriik (actually running on a temporary DigitalOcean VPS) and it instantly synced three last activities which stuck on tapiriik.com for 4 days already

@cpfair there is an issue here for sure. Any chance you'll look into it in near future? May I ask you to at least estimate the chances you'll have time to investigate? If you are busy right now, I'll invest additional time to safely set everything up on my production server (my current attempt was quick & dirty).

**Comment by RobDickson** [comment]
I haven't had a successful sync for over a week and have only have about half-a-dozen successful syncs this year. Without me doing anything, Tapiriik will occasionally decide to synchronise with Garmin Connect and most of the missing activities will suddenly appear in Strava. Nothing will then happen for a week or two.

I can't offer to help with coding, but let me know if I can do anything else to help.

**Comment by mishamsk** [comment]
@RobDickson right after posting my last comment, I've changed the setting on Garmin connector to include private activities (all of my activities are for "My Connections" only, not public and not private) + set the minimum date to sync activities to the date of my last comment. Since that change everything is syncing almost immediately! I do not know if the settings somehow changed the behavior or 3 activities which I've synced through local installation somehow unblocked smth. Nevertheless it is working properly now.

**Comment by RobDickson** [comment]
Thanks for the suggestion.

Changing the date allowed Tapiriik to synchronise most of my missing activities. However, it didn't allow it to synchronise any subsequent ones. To do this, I had to change the date again, which synchronised most of the missing activities, but, again, no ones after that.

There seems little point in having an automatic sync if I have to log on to Tapiriik to force a sync. I have just changed the date again. I expect this will allow today's two rides to sync, but not this evening's swim (which I haven't done yet).

**Comment by RobDickson** [comment]
That's it. Tapiriik tries to synchronise with Garmin every hour, but fails to do so until it's nudged into action by changing the date to synchronise from. It synchronises only once, then goes back to failing every hour until I change the date to synchronise from again. This seems to have happened since Christmas.

I hope somebody can fix this soon......

**Comment by mishamsk** [comment]
@RobDickson sorry to hear that, it is been very reliable for me since then. Syncs in an hour 100% of times for the last week...

**Comment by RobDickson** [comment]
It started working again last week, but has stopped again now. Does anybody know when this will be fixed permanently?

**Comment by RobDickson** [comment]
It started working about for about a week up until Monday of this week. It hasn't worked since then.

**Comment by AndrejWalk** [comment]
For me It doesn't work since Sept. 2016. No GC activities since 12 Sep 2016 appear in my list. Change password, remove and add GC. No change. 
I check number of synced activities: 1024 ??
Activities in GC: 1150
Accounts: Dropbox <- Garmin Connect <->  Endomondo -> Dropbox

**Comment by RobDickson** [comment]
Andrej, does it work for you now?

**Comment by AndrejWalk** [comment]
No, it doesn't work yet. It synchronize only 1000 activities. But I have 1274 activities in GC.
For me, nothing after 12. Sep. 2016.

**Comment by RobDickson** [comment]
@AndrejWalk It seems the number of activities is the problem for you. Why not change the start date to 01/09/16 and, once that has worked, change it to 01/06/17?

**Comment by gustavosoares** [comment]
yep, not syncing for me either...
